pages/users/user.js

```diff
@@ -0,0 +1,255 @@
+
+    // Ensure current user and displayed user exists, then make sure each user has groups to compare. if any condition isn't met, return false.
+    if (!currentUser || !displayedUser || !currentUser.relationships.groups.data.length || !displayedUser.relationships.groups.data.length) {
+      return false
+    }
+
+    // Get a list of group ids for both current user and displayed user.
+    const currentUserGroups = currentUser.relationships.groups.data.map(group => group.id)
+    const displayedUserGroups = displayedUser.relationships.groups.data.map(group => group.id)
+
+    // If any of the current user's groups are shared with the display user, return true.
+    return currentUserGroups.some(group => displayedUserGroups.includes(group))
+  }
+
+  get renderUserProfile() {
+    const {
+      displayedUser,
+    } = this.state
+
+    const {
+      avatar,
+      bio,
+      email,
+      playedGames,
+      username,
+      wantToPlayGames,
+    } = displayedUser.attributes
+
+    const displayedUserSharesGroup = this.currentUserSharesGroup()
+
+    return (
+      <React.Fragment>
+        <div className="page-content">
+          <div className="user-details">
+            <div className="user-details-card">
+              <img
+                className="user-image"
+                src={avatar ? `/api/users/${displayedUser.id}/avatar` : `//api.adorable.io/avatars/500/${encodeURIComponent(displayedUser.id)}`}
+                alt="User Avatar" />
+
+              <div className="user-name">
+                <h4>{username}</h4>
+              </div>
+
+              <menu
+                className="compact"
+                type="toolbar">
+
+                {displayedUserSharesGroup && (
+                  <a
+                    className="button small success"
+                    href={`mailto:${email}`}>
+                    Message
+                  </a>
+                )}
+
+                <button className="danger small">
+                  block
+                </button>
+
+              </menu>
+            </div>
+          </div>
+          <div className="user-bio">
+            <h4>About {username}</h4>
+            <div className="bio-content">
+              { bio || 'Blah blash blah I\'m a user bio. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.'}
+            </div>
+          </div>
+          <div className="user-favorite-games">
+            <h4>{username}'s favorite games</h4>
+            <ul className="gamelist">
+              {playedGames.map(game => (
+                <li key={game.replace(' ', '')}>{game}</li>
```

Comment:
We don't need to replace it, but `key` doesn't have to be a string without spaces. It can be literally any string.

Reply:
When we come back to this after alex finishes game lists, I'll take it out.